### PR TITLE
added solution to the task

### PR DESCRIPTION
**Pull request by yanna-stepanova**

### CODE REVIEW

File: src/main/java/com/yanna/stepanova/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String HEADER_NAME = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/security/JwtAuthenticationFilter.java]: You can use HttpHeaders.AUTHORIZATION Spring constant

File: src/main/java/com/yanna/stepanova/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    private static final String CHECK_WORD = "Bearer ";
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_HEADER = "Bearer ";
```

File: src/main/java/com/yanna/stepanova/controller/BookController.java [hunk 3]
```diff
@@ -58,8 +59,10 @@ public List<BookDto> getAllByAuthor(@RequestParam String author) {
     @GetMapping
     @Operation(summary = "Get all books in parts",
             description = "Get all the books in parts + using sorting")
-    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
-        return bookService.getAll(pageable);
+    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable,
```
Comment (Elena-Bruyako) [inline src/main/java/com/yanna/stepanova/controller/BookController.java]: Add `@PreAuthorize` where is needed under the BookController endpoints. 
You don't need to provide `authentication.getName()` to getAll books.

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Your build failed. Please, check it before sending the PR.

File: src/main/java/com/yanna/stepanova/controller/BookController.java (hunk 3 above)
Comment (yanna-stepanova) [reply src/main/java/com/yanna/stepanova/controller/BookController.java]: For "role_admin" @PreAuthorize was added. Should I add it for "role_user"? Now endpoints are by default for everyone where @PreAuthorize is missing.

**Review by MAMentorFX — APPROVED** [review]

**Pull request merged**